### PR TITLE
Fix a11y dropdown navigation

### LINKED ISSUE #9561: TalkBack incorrectly reading links in nested spans
**Describe the bug**
When using TalkBack on Android, links with role text and nested spans aren't read as one item as intended to be. Content is also repeated in additional blocks/gestures.

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to https://www.bbc.com/portuguese/podcasts/p07r3r3t
2. Navigate to the 'O podcast está disponível em' header on TalkBack
3. Swipe to the list on TalkBack
4. TalkBack should read 'Spotify, Que Historia, externo, link'
5. Swipe through the links and TalkBack should read an additional 'Spotify', and the text highlighted
6. Swipe again and TalkBack will read out 'Que Historia' as well as highlight a separate 'block' on the screen.

**Expected behaviour**
TakBack should only read out 'Spotify, Que Historia, externo, link' before moving on to the next link.

**Screenshots**
image

**Smartphone (please complete the following information):**

- Device: Samsung S10
- OS: Android 11
- Browser: Chrome
- Version: 94.0.4606.85

**Testing notes**

- [x] This bug fix is expected to need manual testing.

**Checklist**
- [x] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

### PR DESCRIPTION
**Pull request by DarioR01**
Resolves [#9561](https://github.com/bbc/simorgh/issues/9561)

Fix an a11y issue taht caused Talkback to read out comma as "comma" instead of taking a break.
Fix an a11y issue where nested span with a parent having role=text gets read with multiple gestures (it should only take one gesture).

**Code changes:**

- Created Storybook about [dropdown Navigation](https://5d28eb5ee163f6002046d6fb-snibgjppbx.chromatic.com/?path=/story/components-navigation-canonical--dropdown-navigation)
- Concatenated the comma with previous values displayed
- added aria-labelledby to Anchor element, and id to span elements with role=text

---

- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: packages/components/psammead-navigation/CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,8 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
-| 9.2.14 | [PR#4587](https://github.com/bbc/psammead/pull/4587) Fix TalkBack reading nested spans incorrectly |
+| 9.2.14 | [PR#4599](https://github.com/bbc/psammead/pull/4599) Fix TalkBack reading nested spans incorrectly, and reading ',' as 'comma' in Dropdown Navigation|
+| 9.2.14 | [PR#4587](https://github.com/bbc/psammead/pull/4587) Fix TalkBack reading nested spans incorrectly in Scrollbar Navigation|
```
Comment (greenc05) [inline packages/components/psammead-navigation/CHANGELOG.md:7]: Twice?

File: packages/components/psammead-navigation/CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,8 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
-| 9.2.14 | [PR#4587](https://github.com/bbc/psammead/pull/4587) Fix TalkBack reading nested spans incorrectly |
+| 9.2.14 | [PR#4599](https://github.com/bbc/psammead/pull/4599) Fix TalkBack reading nested spans incorrectly, and reading ',' as 'comma' in Dropdown Navigation|
```
Comment (greenc05) [inline packages/components/psammead-navigation/CHANGELOG.md]: Wrong version number?

File: packages/components/psammead-navigation/src/DropdownNavigation/__snapshots__/index.test.jsx.snap [hunk 3]
```diff
@@ -980,54 +1037,79 @@ exports[`Dropdown navigation should render correctly when open 1`] = `
...
+            Entertainment
+          </span>
         </a>
       </li>
       <li
         class="emotion-4 emotion-5"
         role="listitem"
       >
         <a
+          aria-labelledby="dropdownNavigation-Most popular"
```
Comment (greenc05) [inline packages/components/psammead-navigation/src/DropdownNavigation/__snapshots__/index.test.jsx.snap]: Id text looks like it's missing a dash between Most and popular?

Comment (greenc05) [reply packages/components/psammead-navigation/src/DropdownNavigation/__snapshots__/index.test.jsx.snap]: Not that this is needed here, only needed when we have current page

**Comment by greenc05** [comment]
Thanks Dario

File: packages/components/psammead-navigation/CHANGELOG.md (hunk 2 above)
Comment (DarioR01) [reply packages/components/psammead-navigation/CHANGELOG.md]: Yes it is, thank you 👍

File: packages/components/psammead-navigation/src/DropdownNavigation/__snapshots__/index.test.jsx.snap (hunk 3 above)
Comment (DarioR01) [reply packages/components/psammead-navigation/src/DropdownNavigation/__snapshots__/index.test.jsx.snap]: Now code is smart enough to replace spaces with dashes 😄

**Review by MuhammadAHussain — APPROVED** [review]
LGTM 👍

**Review by amoore108 — APPROVED** [review]

**Pull request merged**